src/app/Providers/components/AddEditProviderModal/AddEditProviderModal.tsx

```diff
@@ -195,6 +199,39 @@ export const AddEditProviderModal: React.FunctionComponent<IAddEditProviderModal
     ? createProviderMutation
     : patchProviderMutation;
 
+  const [hostname, setHostname] = React.useState(fields?.hostname ? fields.hostname.value : '');
+
+  function assertIsCertificate(
+    certificate: ITLSCertificate
+  ): asserts certificate is ITLSCertificate {
```

Comment:
whoa!! I was not aware of this TypeScript feature, I really wish I had discovered this sooner! That's really nice. I think we could use it in a few places where we have logic based on providerType and mappingType with a lot of `as` assertions. Future cleanup :)